source/transactions-convenient-api/transactions-convenient-api.md

```diff
@@ -128,11 +129,23 @@ This method should perform the following sequence of actions:
 6. If the callback reported an error:
     1. If the ClientSession is in the "starting transaction" or "transaction in progress" state, invoke
         [abortTransaction](../transactions/transactions.md#aborttransaction) on the session.
+
     2. If the callback's error includes a "TransientTransactionError" label and the elapsed time of `withTransaction` is
-        less than 120 seconds, jump back to step two.
+        less than 120 seconds, calculate the backoff value to be
+        `jitter * min(BACKOFF_INITIAL * (1.25**retry), BACKOFF_MAX)` where:
+
+        1. jitter is a random float between \[0, 1)
+        2. retry is one less than the number of times Step 2 has been executed since Step 1 was executed
+        3. BACKOFF_INITIAL is 1ms
+        4. BACKOFF_MAX is 500ms
+
+        If the time elapsed thus far plus the backoff value would not exceed 120 seconds, then sleep for the backoff
+        value and jump back to step two, otherwise, raise last known error.
+
```

Comment:
I'm not sure why the spec doesn't mention CSOT.  We should be factoring timeoutMS into this algorithm, I think it would look something like:

- calculate the backoffMS value
- if timeoutMS is set and remainingTimeMS < backoffMS: raise last known error
- if timeoutMS is not set and elapsed time + backoff > 120, raise last known error
- sleep for backoffMS and return to step 2

Reply:
thanks for poitning that out -- fixed in [42e4d94](https://github.com/mongodb/specifications/pull/1851/commits/42e4d94146cee87c3ef66539479185c3d417803c)